packages/amplication-server/src/core/moduleDto/moduleDto.service.ts

```diff
@@ -227,7 +228,7 @@ export class ModuleDtoService extends BlockTypeService<
         ...args,
         data: {
           ...args.data,
-          properties: [],
+          properties: (args.properties as unknown as JsonArray) ?? [],
```

Comment:
⚠️ pay attention to this casting as it will hide potential type mismatches.
Type casting should be avoided in typescript projects

Reply:
that is true. 
the BlockCreateInput uses a specific implementation of [key: string]: JsonValue; which causes a type conflict here. 
If you think you have a quick workaround, please suggest one, otherwise it will not make sense to make a big change to fix it now

Reply:
@overbit  please note that because of the `[key: string]: JsonValue;` BlockCreateInput anyway accepts all types, and the type validation is done on the previous layer per specific block type